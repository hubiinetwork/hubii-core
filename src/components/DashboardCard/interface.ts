export interface IDashboardCardProps {
  title: string;
  href: string;
  iconType: string;
}
