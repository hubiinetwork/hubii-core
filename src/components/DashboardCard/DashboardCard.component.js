import React from 'react';
import PropTypes from 'prop-types';
import {
  Wrapper,
  AntdIcon,
  CustomIcon as Icon,
  Title,
} from './DashboardCard.style';

/**
 * This component shows app's features as options on the main screen.
 */

const DashboardCard = ({ className, iconType, title, iconSrc }) => (
  <Wrapper className={className}>
    {iconSrc ? <Icon src={iconSrc} /> : <AntdIcon type={iconType} />}
    s
    <Title>{title}</Title>
  </Wrapper>
  );

DashboardCard.propTypes = {
  /**
   * title to show on dashboard card.
   */
  title: PropTypes.string.isRequired,

  /**
   * antd icon to show on dashboard card with title.
   */
  iconType: PropTypes.string,

  /**
   * location of icon to show on dashboard card with title.
   */
  iconSrc: PropTypes.string,

  /**
   * styled-components classname
   */
  className: PropTypes.string,
};

export default DashboardCard;
