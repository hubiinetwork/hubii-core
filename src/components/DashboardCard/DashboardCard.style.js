import { Card } from 'antd';
import styled from 'styled-components';

const StyledCard = styled(Card)`
  min-width: 370px;
  height: 113px;
  border-radius: 8px;
  text-align: left;
  padding: 0px 29px;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.1);
  background-color: ${({ theme }) => theme.palette.primary4};
  border: 1px solid ${({ theme }) => theme.palette.primary4} !important;
  &:hover {
    background-color: ${({ theme }) => theme.palette.primary5};
    border: 1px solid ${({ theme }) => theme.palette.secondary3} !important;
  }
  &:focus {
    text-decoration: none;
  }
`;

const IconSpan = styled.span`
  top: 5px;
  font-size: 48px;
  position: relative;
  color: ${({ theme }) => theme.palette.info};
`;

const TitleSpan = styled.span`
  top: -7px;
  font-size: 18px;
  font-weight: 500;
  line-height: 21px;
  position: relative;
  padding-left: 20px;
  color: ${({ theme }) => theme.palette.light};
`;

const Wrapper = styled.div`
  font-size: 20px;
  margin-top: 20px;
  margin-right: 15px;
  display: inline-flex;
  cursor: pointer;
`;
export { Wrapper, StyledCard, IconSpan, TitleSpan };
