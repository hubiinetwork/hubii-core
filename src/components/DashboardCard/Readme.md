Basic DashboardCard:

```js
<DashboardCard iconType="wallet" to="/" title="Wallet Manager" />
```

Basic DashboardCard:

```js
<DashboardCard iconType="setting" to="/" title="Settings" />
```
