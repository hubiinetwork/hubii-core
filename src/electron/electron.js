
/* global mainWindow */
import { initSplashScreen } from '@trodi/electron-splashscreen';
import { app, Menu, ipcMain } from 'electron';
import log from 'electron-log';
import { autoUpdater } from 'electron-updater';
import windowStateKeeper from 'electron-window-state';
import path from 'path';
import isDev from 'electron-is-dev';
import { registerWalletListeners } from './wallets';
import setupDevToolsShortcut from './dev-tools';

if (process.env.NODE_ENV === 'development') {
  // eslint-disable-next-line global-require
  require('electron-reload')(__dirname, {
    electron: path.join(`${__dirname}/../../`, 'node_modules', '.bin', 'electron'),
  });
}

const version = app.getVersion();

function createWindow() {
  const template = [
    {
      label: 'Application',
      submenu: [
        { label: `Version: ${version}`, enabled: false },
        { type: 'separator' },
        { label: 'Quit', accelerator: 'Command+Q', click() { app.quit(); } },
      ],
    },
    {
      label: 'Edit',
      submenu: [
        { label: 'Undo', accelerator: 'CmdOrCtrl+Z', selector: 'undo:' },
        { label: 'Redo', accelerator: 'Shift+CmdOrCtrl+Z', selector: 'redo:' },
        { type: 'separator' },
        { label: 'Cut', accelerator: 'CmdOrCtrl+X', selector: 'cut:' },
        { label: 'Copy', accelerator: 'CmdOrCtrl+C', selector: 'copy:' },
        { label: 'Paste', accelerator: 'CmdOrCtrl+V', selector: 'paste:' },
        { label: 'Select All', accelerator: 'CmdOrCtrl+A', selector: 'selectAll:' },
      ],
    },
  ];

  Menu.setApplicationMenu(Menu.buildFromTemplate(template));

  const windowOptions = {
    width: 1250,
    height: 780,
    show: false,
    icon: process.platform === 'linux' && path.join(__dirname, '../icon.png'),
    webPreferences: {
      preload: path.join(__dirname, 'preload.js'),
      webSecurity: false,
    },
  };
  // Load the previous state with fallback to defaults
  const mainWindowState = windowStateKeeper({
    defaultWidth: windowOptions.width,
    defaultHeight: windowOptions.height,
    fullScreen: false,
    maximize: false,
  });

  global.mainWindow = initSplashScreen({
    windowOpts: { ...windowOptions, ...mainWindowState },
    templateUrl: path.join(__dirname, '../public/images/splashscreen.svg'),
    delay: 0,
    splashScreenOpts: {
      height: 340,
      width: 340,
      transparent: true,
    },
  });

  mainWindowState.manage(mainWindow);

  mainWindow.loadURL(isDev ? 'http://localhost:3000' : `file://${path.join(__dirname, '../index.html')}`);
  if (isDev) {
    mainWindow.webContents.openDevTools();
    // Need to require this globally so we can keep it as a
    // dev-only dependency
    const {
      default: installExtension,
      REACT_DEVELOPER_TOOLS,
      REDUX_DEVTOOLS,
    } = require('electron-devtools-installer'); // eslint-disable-line global-require
    [REACT_DEVELOPER_TOOLS, REDUX_DEVTOOLS].forEach((extension) => {
      installExtension(extension)
        .then()
        .catch((err) => console.error(`An error occurred loading extension ${name}: `, err)); // eslint-disable-line no-console
    });
  }
  mainWindow.once('show', () => {
    registerWalletListeners(mainWindow);
  });
  mainWindow.on('closed', () => {
    app.quit();
  });
  return mainWindow;
}

function setupAutoUpdater() {
  autoUpdater.logger = log;
  autoUpdater.logger.transports.file.level = 'info';

  autoUpdater.on('update-downloaded', () => {
    mainWindow.webContents.send('update-downloaded');
  });

  autoUpdater.on('error', (err) => {
    log.error('auto updater:', err);
  });

  autoUpdater.checkForUpdatesAndNotify();

  ipcMain.on('install-new-release', () => {
    autoUpdater.quitAndInstall();
  });
}

const shouldQuit = app.makeSingleInstance(() => {
  // Someone tried to run a second instance, we should focus our window.
  if (mainWindow) {
    if (mainWindow.isMinimized()) mainWindow.restore();
    mainWindow.focus();
  }
});

if (shouldQuit) {
  app.quit();
} else {
  app.on('ready', () => {
    createWindow();
    setupAutoUpdater();
    setupDevToolsShortcut();
  });
}


app.on('activate', () => {
  if (mainWindow === null) {
    createWindow();
  }
});

process.on('unhandledRejection', (err) => {
  log.error(err);
});
process.on('uncaughtException', (err) => {
  log.error(err);
});
