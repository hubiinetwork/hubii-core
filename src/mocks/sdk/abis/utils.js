const fs = require('fs');
const contractPath = process.env.CONTRACT_PATH;
const path = `${contractPath}/build/addresses.json`;
const addresses = JSON.parse(fs.readFileSync(path));

export function mockAbi(contract) {
  const abi = require(`nahmii-sdk/lib/abis/${contract}`);//eslint-disable-line
  abi.networks[3] = {
    address: addresses.networks['ganache-cli'][contract],
  };
  return abi;
}
