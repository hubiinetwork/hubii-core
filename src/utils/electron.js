import path from 'path';

export function getAbsolutePath(relativePath) {
  // eslint-disable-next-line global-require
  const rootPath = process.env.NODE_ENV === 'production' ? path.join(require('electron').remote.app.getAppPath(), 'build') : '/';
  if (process.env.NODE_ENV === 'test') {
    return path.posix.join(rootPath, relativePath);
  }
  return path.join(rootPath, relativePath);
}
