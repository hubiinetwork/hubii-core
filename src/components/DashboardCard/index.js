import DashboardCard from './DashboardCard.component';
export default DashboardCard;
