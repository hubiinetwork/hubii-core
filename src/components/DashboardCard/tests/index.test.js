import React from 'react';
import { shallow } from 'enzyme';
import { getAbsolutePath } from 'utils/electron';

import DashboardCard from '../DashboardCard.component';

describe('<DashboardCard />', () => {
  it('should render correctly when passed an antd icon type', () => {
    const wrapper = shallow(
      <DashboardCard iconType="setting" title="Settings" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly when passed an icon src', () => {
    const wrapper = shallow(
      <DashboardCard iconSrc={getAbsolutePath('public/images/dex-icon-green.png')} title="DEX" />);
    expect(wrapper).toMatchSnapshot();
  });
});
