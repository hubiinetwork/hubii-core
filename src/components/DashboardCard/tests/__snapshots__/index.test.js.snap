// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardCard /> should render correctly when passed an antd icon type 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashboardCard
    iconType="setting"
    title="Settings"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <DashboardCard.style__AntdIcon
          type="setting"
/>,
        "s",
        <DashboardCard.style__Title>
          Settings
</DashboardCard.style__Title>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "type": "setting",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "s",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Settings",
        },
        "ref": null,
        "rendered": "Settings",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DashboardCard.style__AntdIcon
            type="setting"
/>,
          "s",
          <DashboardCard.style__Title>
            Settings
</DashboardCard.style__Title>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "type": "setting",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "s",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Settings",
          },
          "ref": null,
          "rendered": "Settings",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<DashboardCard /> should render correctly when passed an icon src 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashboardCard
    iconSrc="\\\\public\\\\images\\\\dex-icon-green.png"
    title="DEX"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <DashboardCard.style__CustomIcon
          src="\\\\public\\\\images\\\\dex-icon-green.png"
/>,
        "s",
        <DashboardCard.style__Title>
          DEX
</DashboardCard.style__Title>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "src": "\\\\public\\\\images\\\\dex-icon-green.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "s",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "DEX",
        },
        "ref": null,
        "rendered": "DEX",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DashboardCard.style__CustomIcon
            src="\\\\public\\\\images\\\\dex-icon-green.png"
/>,
          "s",
          <DashboardCard.style__Title>
            DEX
</DashboardCard.style__Title>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": "\\\\public\\\\images\\\\dex-icon-green.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "s",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "DEX",
          },
          "ref": null,
          "rendered": "DEX",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
